packages/css/src/components/progress-list/progress-list.scss

```diff
@@ -45,62 +47,119 @@
+    block-size: var(--ams-progress-list-step-heading-4-marker-block-size);
   }
 }
 
 .ams-progress-list__marker-shape {
   aspect-ratio: 1;
-  background-color: var(--ams-progress-list-marker-background-color);
+  background-color: var(--ams-progress-list-step-marker-background-color);
+  border-color: var(--ams-progress-list-step-marker-border-color);
   border-radius: 50%;
-  inline-size: 1em;
+  border-style: var(--ams-progress-list-step-marker-border-style);
+  border-width: var(--ams-progress-list-step-marker-border-width);
+  box-sizing: border-box;
+  display: flex;
+  justify-content: center;
+
+  /*
+  * round() eliminates sub-pixel rendering issues that distort circular markers.
+  * Minor degradation in unsupported browsers is visually unnoticeable.
+  */
+  // stylelint-disable plugin/use-baseline
+  .ams-progress-list--heading-2 & {
+    block-size: var(--ams-progress-list-step-heading-2-shape-block-size);
+    inline-size: var(--ams-progress-list-step-heading-2-shape-inline-size);
+
+    @supports (block-size: round(down, 1px, 1px)) {
+      block-size: round(down, var(--ams-progress-list-step-heading-2-shape-block-size), 2px);
+      inline-size: round(down, var(--ams-progress-list-step-heading-2-shape-inline-size), 2px);
+    }
+  }
+
+  .ams-progress-list--heading-3 & {
+    block-size: var(--ams-progress-list-step-heading-3-shape-block-size);
+    inline-size: var(--ams-progress-list-step-heading-3-shape-inline-size);
+
+    @supports (block-size: round(down, 1px, 1px)) {
+      block-size: round(down, var(--ams-progress-list-step-heading-3-shape-block-size), 2px);
+      inline-size: round(down, var(--ams-progress-list-step-heading-3-shape-inline-size), 2px);
+    }
+  }
+
+  .ams-progress-list--heading-4 & {
+    block-size: var(--ams-progress-list-step-heading-4-shape-block-size);
+    inline-size: var(--ams-progress-list-step-heading-4-shape-inline-size);
+
+    @supports (block-size: round(down, 1px, 1px)) {
+      block-size: round(down, var(--ams-progress-list-step-heading-4-shape-block-size), 2px);
+      inline-size: round(down, var(--ams-progress-list-step-heading-4-shape-inline-size), 2px);
+    }
+  }
+  // stylelint-enable plugin/use-baseline
+
+  .ams-progress-list__step--completed & {
+    background-color: var(--ams-progress-list-step-completed-marker-background-color);
+    border: var(--ams-progress-list-step-completed-marker-border);
+  }
+
+  .ams-progress-list__step[aria-current] & {
+    background-color: var(--ams-progress-list-step-current-marker-background-color);
+    border-color: var(--ams-progress-list-step-current-marker-border-color);
+  }
+}
+
+.ams-progress-list__marker-icon {
+  align-self: center;
+  inline-size: 100%;
 }
 
 // Connector
 
 .ams-progress-list__connector {
   align-self: stretch;
-  background-color: var(--ams-progress-list-connector-background-color);
+  background-color: var(--ams-progress-list-step-connector-background-color);
```

Comment:
This connector has a background colour, others reset it and have a border. Can we use one of both in all situations? This will probably simplify it a bit further, and is more flexible for the future.

Also, we can’t be sure that the background will be white, so we cannot use that to reset a colour.